lib/checks.js

```diff
@@ -113,21 +114,48 @@ var categories = function(checkName) {
   return [checkCategoryMapping[checkName] || "Style"];
 };
 
-var remediationPoints = function(checkName, message) {
-  if (checkName === "complexity") {
+// Here Be Dragons: this function extracts the relevant value that triggered the issue for
+// checks in the Complexity category. Unfortunately, these values are not available in a
+// structured way, so we extract them from strings. That means that any check categorized
+// as Complexity MUST have a rule here to extract value.
+//
+// If a matching string segment cannot be found, `null` will be returned.
+var messageMetricValue = function(message) {
+  var match = null;
+  switch (message.ruleId) {
+    case "complexity":
+      match = message.message.match(/complexity of (\d+)/);
+      break;
+    case "max-statements":
+      match = message.message.match(/too many statements \((\d+)\)/);
+      break;
+  }
+  if (null !== match) {
+    return parseInt(match[1], 10);
+  }
+  return null;
+};
+
+var metricThreshold = function(ruleId, eslintConfig) {
+   return eslintConfig.rules[ruleId][1];
+};
+
+var remediationPoints = function(checkName, message, eslintConfig) {
```

Comment:
On the one hand: this means finally adding some tests for this engine, which makes me happy. On the other hand, this means I'm responsible for setting up the testing setup on this engine, which makes me sad.

But I agree it's probably worth testing stuff here. I'll take a crack at it.